sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -242,6 +247,10 @@ public FieldType type() {
     public abstract TypeName getTypeName();
     // For container types (e.g. ARRAY), returns the type of the contained element.
     @Nullable public abstract FieldType getComponentType();
+    // For MAP type, returns the type of the key element.
+    @Nullable public abstract FieldType getComponentKeyType();
+    // For MAP type, returns the type of the value element.
+    @Nullable public abstract FieldType getComponentValueType();
```

Comment:
to me it looks like both `getComponentValueType()` and `getComponentType()` mean the same thing, i.e. they both describe the type of the value in the container. Keep just one of them?

Reply:
I would prefer to have different fields for MAP and ARRAY, if it doesn't cause significant performance issue.